### ISSUE TITLE
Plugin stopped working a and cannot be anymore installed

### ISSUE THREAD

**Issue by roberto-cisternino** [Chris2011/minifierbeans#108]
I was using your plugin successfully then stopped working the minification "Error on CLI execution" and even reinstalling the plugin is impossible because the google closuse compiler cannot be extracted after install.
I stopped Bitdefender but error was the same.

**Comment by Chris2011** [comment]
Related to #104

The workaround is to extract the custom-packages.zip manually and please take care that you have the latest custom-packages version which is 1.5.1 which you can find in the package.json. I'm thinking about another solution but the workaround should work. And the end path where the CLIs should be is %user%/.netbaens/minifierbeans/custom-package/*.* so make sure that there is no custom-packages/custom-packages/...

**Comment by roberto-cisternino** [comment]
Thanks !

**Comment by Chris2011** [comment]
If you want to help me, figure out the problem, just play around. I wanted to create another solution but this takes more time than maybe identifing the bug. So it would be very cool, if you want, to play around to figure out what the problem is to help me here.

**Comment by roberto-cisternino** [comment]
> If you want to help me, figure out the problem, just play around. I wanted to create another solution but this takes more time than maybe identifing the bug. So it would be very cool, if you want, to play around to figure out what the problem is to help me here.

I would be happy but I am currently in trouble as well... also I can't understand why it stopped working ... a mistery.

Btw, I tries to unzip but once I reload Netbeans it is trying again to unzip the package and a notification appear with a link to retry the unzip and this error is available in the IDE log:

java.lang.NullPointerException: Cannot invoke "java.util.concurrent.Future.get(long, java.util.concurrent.TimeUnit)" because "task" is null
	at io.github.chris2011.netbeans.minifierbeans.javascript.JsMinify.jsMinify(JsMinify.java:125)
	at io.github.chris2011.netbeans.minifierbeans.javascript.JsMinify.lambda$execute$0(JsMinify.java:75)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1419)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:287)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2034)

**Comment by Chris2011** [comment]
Yeah, because it couln't find the cmd file, which seems you have an old custom-package.zip archive. Please check the version as mentioned. above and also try the workaround.

**Comment by roberto-cisternino** [comment]
> Yeah, because it couln't find the cmd file, which seems you have an old custom-package.zip archive. Please check the version as mentioned. above and also try the workaround.

Yes Chris, I checked the version is 1.5.1

**Comment by Chris2011** [comment]
Please check the path of the unziped package. It needs to be in just one custom-package folder. If you see a custom-packages/custom-packages, this is wrong.

**Comment by roberto-cisternino** [comment]
> Please check the path of the unziped package. It needs to be in just one custom-package folder. If you see a custom-packages/custom-packages, this is wrong.

Ok, now I have all inside custom-package as follows:
C:\Users\...\.netbeans\minifierbeans\custom-package
    \node_modules
    ...

But when I try to minify a js I obtain this notification:
![error_while_extracting](https://github.com/Chris2011/minifierbeans/assets/29631792/98c4b392-7526-466e-a654-7d6145e903c2)

**Comment by Chris2011** [comment]
I think I misspacked smth. First the path is wrong which needs to be custom-packages and not custom-package. For the overall extracting problem, I will dive into it and add logging for myself.

**Comment by roberto-cisternino** [comment]
> I think I misspacked smth. First the path is wrong which needs to be custom-packages and not custom-package. For the overall extracting problem, I will dive into it and add logging for myself. I also mentioned it incorrect in the other post "custom-package" which is wrong.

Ok. Thanks Chris... it was my fault as I missed the final s.  Now it works fine !
So the reason for stopping working remains unknown, but the unzip process was starting the unzip without creating the custom-packages folder, but then was interrupted and unable to unzip the sub-folders.  Anyway, it looks it was unable to create the custom-packages folder before unzipping.

**Comment by Chris2011** [comment]
We had 2 problems. First was that I missed to zip the custom-packages folder as full one so inside the custom-packages.zip the folder custom-packages needs to be there. Second problem is that it seems to have problems to create folders as you said it correct. This is the exception, that I get

```
java.nio.file.NoSuchFileException: C:\Users\clenz\.netbeans\minifierbeans\custom-packages\google-closure-compiler
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:371)
	at java.base/java.nio.file.Files.createFile(Files.java:648)
	at io.github.chris2011.netbeans.minifierbeans.util.FileUtils.extractArchive(FileUtils.java:277)
	at io.github.chris2011.netbeans.minifierbeans.util.FileUtils.lambda$downloadFile$0(FileUtils.java:235)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1420)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:287)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2035)
```

But I dunno why this happens now. Maybe a problem with rights? I can remember this problem with linux: #99 but this was already for the files that are exist. I will try to find a fix for that.

**Comment by Chris2011** [comment]
Ok the problem seems the compression algorithm (I guess). The last times it seems that I used 7z with compression strength 9 - Ultra. The last time (couple of days ago) I just used windows zip to create the archive. After I played with 7z it worked again but the decompression took veeeeeeeeery long.

**Comment by Chris2011** [comment]
So all in all, the first problem was, that I zipped the archive with a wrong compression strength. I just used windows zipping, before I used 7z with highest compression strength (9 - Ultra). The second problem was, that I removed the custom-packages folder from the zip and added the files to the root path which is wrong.

This is fixed in custom-packages 1.5.2 and it should work now out of the box, after opening NetBeans. If not, please empty the folder .netbeans/minifierbeans and restart NetBeans.

With Minifierbeans 3.5.1 I also increased the performance of extracting the archive.
